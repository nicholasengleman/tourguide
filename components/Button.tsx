import { Pressable, StyleSheet, Text, View } from 'react-native';

interface ButtonProps {
  text: string;
  number: number;
  icon?: string;
  onPress?: () => void;
}

export const Button: React.FC<ButtonProps> = ({
  text,
  icon,
  number,
  onPress,
}) => {
  return (
    <View>
      <Pressable
        style={({ pressed }) => [
          styles.button,
          pressed ? styles.buttonPressed : null,
        ]}
        onPress={onPress}
      >
        <Text>{text}</Text>
        <View style={styles.number}>
          <Text style={styles.numberText}>{number}</Text>
        </View>
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: 'lightgrey',
    borderRadius: 20,
    paddingHorizontal: 15,
    paddingVertical: 5,
    flexDirection: 'row',
    gap: 5,
    alignItems: 'center',
    fontFamily: 'Roboto-Regular',
  },
  buttonPressed: {
    backgroundColor: 'blue',
  },
  number: {
    color: 'black',
    backgroundColor: 'white',
    width: 23,
    height: 23,
    borderRadius: 10,
    overflow: 'hidden',
    alignItems: 'center',
    justifyContent: 'center',
    marginLeft: 5,
    borderWidth: 2,
    shadowColor: 'black',
    shadowOffset: { width: 5, height: 5 },
    shadowRadius: 6,
    shadowOpacity: 1,
    elevation: 5,
  },
  numberText: {
    fontFamily: 'Roboto-Bold',
  },
});
