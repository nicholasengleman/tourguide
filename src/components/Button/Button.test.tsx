import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { Button } from "./Button";

describe("<Button />", () => {
  it("renders the button text", () => {
    const { getByText } = render(<Button text="Click me" number={5} />);
    expect(getByText("Click me")).toBeTruthy();
  });

  it("renders the button number", () => {
    const { getByText } = render(<Button text="Click me" number={5} />);
    expect(getByText("5")).toBeTruthy();
  });

  it("calls onPress when the button is pressed", () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button text="Click me" number={5} onPress={onPressMock} />
    );
    fireEvent.press(getByText("Click me"));
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it("applies pressed styles when the button is pressed", () => {
    const { getByText, toJSON } = render(<Button text="Click me" number={5} />);
    const button = getByText("Click me").parent;

    // Before pressing
    expect(toJSON()).toMatchSnapshot("unpressed");

    // Simulate press
    fireEvent(button, "pressIn");
    expect(toJSON()).toMatchSnapshot("pressed");
  });

  it("doesn't call onPress when onPress prop is not provided", () => {
    const onPressMock = jest.fn(); // Create a mock function even though it won't be used
    const { getByText } = render(<Button text="Click me" number={5} />);
    fireEvent.press(getByText("Click me"));
    expect(onPressMock).not.toHaveBeenCalled();
  });

  // This test is not really necessary since the icon prop is not being used
  it("does not render an icon when icon prop is not provided", () => {
    const { queryByTestId } = render(<Button text="Click me" number={5} />);
    expect(queryByTestId("icon")).toBeNull();
  });

  // This test is not really necessary since the icon prop is not being used
  it("renders an icon when icon prop is provided", () => {
    const { queryByTestId } = render(
      <Button text="Click me" number={5} icon="some-icon-name" />
    );
    expect(queryByTestId("icon")).toBeNull(); // Still null because it's not rendered
  });

  it("matches snapshot", () => {
    const { toJSON } = render(<Button text="Click me" number={5} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
