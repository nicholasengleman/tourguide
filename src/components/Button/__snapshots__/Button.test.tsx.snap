// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> applies pressed styles when the button is pressed: pressed 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 20,
          "flexDirection": "row",
          "fontFamily": "Roboto-Regular",
          "gap": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 5,
        },
        null,
      ]
    }
  >
    <Text>
      Click me
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "color": "black",
          "elevation": 5,
          "height": 23,
          "justifyContent": "center",
          "marginLeft": 5,
          "overflow": "hidden",
          "shadowColor": "black",
          "shadowOffset": {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
          "width": 23,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Roboto-Bold",
          }
        }
      >
        5
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Button /> applies pressed styles when the button is pressed: unpressed 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 20,
          "flexDirection": "row",
          "fontFamily": "Roboto-Regular",
          "gap": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 5,
        },
        null,
      ]
    }
  >
    <Text>
      Click me
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "color": "black",
          "elevation": 5,
          "height": 23,
          "justifyContent": "center",
          "marginLeft": 5,
          "overflow": "hidden",
          "shadowColor": "black",
          "shadowOffset": {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
          "width": 23,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Roboto-Bold",
          }
        }
      >
        5
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Button /> matches snapshot 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 20,
          "flexDirection": "row",
          "fontFamily": "Roboto-Regular",
          "gap": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 5,
        },
        null,
      ]
    }
  >
    <Text>
      Click me
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 2,
          "color": "black",
          "elevation": 5,
          "height": 23,
          "justifyContent": "center",
          "marginLeft": 5,
          "overflow": "hidden",
          "shadowColor": "black",
          "shadowOffset": {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
          "width": 23,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Roboto-Bold",
          }
        }
      >
        5
      </Text>
    </View>
  </View>
</View>
`;
